src/client/datascience/notebookStorage/vscNotebookModel.ts

```diff
@@ -89,6 +89,14 @@ export class VSCodeNotebookModel extends BaseNotebookModel {
         this._cells = this.notebookJson.cells || [];
         this._preferredLanguage = cellLanguageService.getPreferredLanguage(this.metadata);
     }
+    public getCells() {
+        return this._cells;
```

Comment:
Please change to `cells` property & add in the base class, the other class used for WebView notebooks has a `cells` property.

Reply:
cells property on other class returns a different type though (ICell []) instead of nbformat.IBaseCell[]. I can change it in this sub class, but changing the return type will ripple elsewhere.

Reply:
Not required.
See comments regarding not requiring this change for native notebooks.
When saving, we call `model.getContent`
& the `getContent` method calls the `getnerateNotebookContent` -> `generateNotebookJson` -> `getJupyterCells`.

In VSC Native notebooks we use `createJupyterCellFromVSCNotebookCell` which gets the raw text from the Text Document.
I.e. ICell is not used in native notebooks when editing/saving (only required for loading - would love to get rid of that as well, but thats a bigger change).